demo/gpu-test6.yaml

```diff
@@ -0,0 +1,48 @@
+# One pod: one initContainer + one container
+# Each asking for shared access to a single GPU
+
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: gpu-test6
+
+---
+apiVersion: resource.k8s.io/v1beta1
+kind: ResourceClaimTemplate
+metadata:
+  namespace: gpu-test6
+  name: single-gpu
+spec:
+  spec:
+    devices:
+      requests:
+      - name: gpu
+        deviceClassName: gpu.example.com
```

Comment:
Right now all of the examples together take up exactly the 8 GPUs exposed by default. I don't think that should block adding new examples though.

Maybe one short-term workaround is to leave the default count exposed by the driver to 8 and override that to 9 for the e2e tests so we can accommodate this new example there? https://github.com/kubernetes-sigs/dra-example-driver/blob/4a1bf91c32d7ccf43e93d7cd6dfaaf7564a7fae2/test/e2e/e2e.sh

I don't think the chart currently exposes a way to override that, so that would also need to be added. I think we can integrate this with tests later though.

@jackfrancis Could you please open an issue to add this to the e2e tests so we don't lose track of that?

Reply:
I'd be happy to add this to the E2E test as part of merging as well if that makes sense?

Reply:
That'd be great!

Reply:
Added E2E matter for this, and updated the hard-coded num devices to 9. I think this should literally work once we build an updated image w/ these changes?

Maybe the larger question is: is this the right way to integrate an initContainer example? Would updating one of the existing examples be easier and clearer in terms of the convenient image representation that describes the current set of six scenarios?

Happy to take whatever approach that is best for keeping the overall set of reference scenarios the cleanest for maintenance, and easiest to present to users.